docker/toolchain.cmake

```diff
@@ -0,0 +1,57 @@
+# default toolchain file for targets, see #1110
+# required so CMAKE_CROSSCOMPILING_EMULATOR is set,
+# as well for embedded systems and other targets.
+#
+# all embedded systems without an OS should set the system name to generic
+# https://cmake.org/cmake/help/book/mastering-cmake/chapter/Cross%20Compiling%20With%20CMake.html
+
+set(CMAKE_SYSTEM_NAME "$ENV{CROSS_CMAKE_SYSTEM_NAME}")
+set(CMAKE_SYSTEM_PROCESSOR "$ENV{CROSS_CMAKE_SYSTEM_PROCESSOR}")
+if(DEFINED ENV{CROSS_TARGET_RUNNER})
+    set(CMAKE_CROSSCOMPILING_EMULATOR $ENV{CROSS_TARGET_RUNNER})
+endif()
+
+# not all of these are standard, however, they're common enough
+# that it's good practice to define them.
+set(prefix "$ENV{CROSS_TOOLCHAIN_PREFIX}")
+set(CMAKE_C_COMPILER "${prefix}gcc")
+set(CMAKE_ASM_COMPILER "${prefix}gcc")
+set(CMAKE_CXX_COMPILER "${prefix}g++")
+set(CMAKE_AR "${prefix}ar")
+set(CMAKE_LINKER "${prefix}ld")
+set(CMAKE_NM "${prefix}nm")
+set(CMAKE_OBJCOPY "${prefix}objcopy")
+set(CMAKE_OBJDUMP "${prefix}objdump")
+set(CMAKE_RANLIB "${prefix}ranlib")
+set(CMAKE_STRIP "${prefix}strip")
+
+# these are cached so any build system that compiled outside of the rust
+# build system, such as a third-party cmake build and install of a shared
+# library, will still work. however, cmake-rs can override these values
+if(DEFINED ENV{CROSS_CMAKE_OBJECT_FLAGS})
+    set(CMAKE_C_FLAGS "$ENV{CROSS_CMAKE_OBJECT_FLAGS}" CACHE INTERNAL "C Compiler options")
+    set(CMAKE_CXX_FLAGS "$ENV{CROSS_CMAKE_OBJECT_FLAGS}" CACHE INTERNAL "C++ Compiler options")
+    set(CMAKE_ASM_FLAGS "$ENV{CROSS_CMAKE_OBJECT_FLAGS}" CACHE INTERNAL "ASM Compiler options")
+endif()
+
+# if cross-compiling, we need to disable where the root path
+# is found and also provide our own sysroot
+set(native ENV{CROSS_CMAKE_NATIVE})
```

Comment:
This can be removed, since the native Dockerfiles don't actually use a custom toolchain file.